mypy/newsemanal/semanal.py

```diff
@@ -1214,7 +1214,11 @@ def prepare_class_def(self, defn: ClassDef, info: Optional[TypeInfo] = None) ->
             #       incremental mode and we should avoid it.
             if '@' not in defn.info._fullname:
                 local_name = defn.info._fullname + '@' + str(defn.line)
-                defn.info._fullname = self.cur_mod_id + '.' + local_name
+                if defn.info.is_named_tuple:
+                    # Module already correctly set for named tuples.
+                    defn.info._fullname += '@' + str(defn.line)
```

Comment:
It's unclear why we need this special case. Even if the module was already correctly set, if it's always `self.cur_mod_id`, I think that it's better to avoid this if statement if it doesn't change behavior.

Reply:
As discussed elsewhere this is (unfortunately) actually needed.